content/en/tracing/guide/instrument_custom_method.md

```diff
@@ -0,0 +1,177 @@
+---
+title: Instrument a custom method to get deep visibility into your business logic
+kind: guide
+disable_toc: true
+further_reading:
+- link: "/tracing/guide/alert_anomalies_p99_database/"
+  tag: "3 mins"
+  text: "Alert on anomalous p99 latency of a database service"
+- link: "tracing/guide/week_over_week_p50_comparison/"
+  tag: "2 mins"
+  text: "Compare a service’s latency to the previous week"
+- link: "/tracing/guide/slowest_request_daily/"
+  tag: "3 mins"
+  text: "Debug the slowest trace on the slowest endpoint of a web service"
+- link: "tracing/guide/"
+  tag: ""
+  text: "All guides"
+---
+_8 minutes to complete_
+
+{{< img src="tracing/guide/custom_span/custom_span_1.png" alt="Analytics View" responsive="true" style="width:90%;">}}
+
+In order to provide you deep visibility into your business logic, Datadog APM allows you to customize the spans that make up your traces based on your needs and implementation. This empowers you to trace any method in your codebase and even specific components within methods. You can use this to optimize and monitor critical areas of your application at the granularity that works for you.
+
+Custom spans are useful when you have methods that perform multiple actions like fetching data from a payments database, sanitizing, processing and verifying the response, and finally returning the results to a customer.
+The database call and sanitization methods are already instrumented automatically, but the processing is entirely tailored to your application. By creating spans for the processing, you can optimize the timing using the APM flamegraph and monitor for errors using Datadog monitors.
+
+**Note**: When using any framework that is integrated with Datadog APM, your web requests, database queries, and cache jobs are automatically traced.
+
+## Instrument a complete method with decorators
+
+**Follow the example to get your code instrumented**.
+
+This examples walks through tracing the entire `BackupLedger.write` method to measure it's execution time and status. `BackupLedger.write` is an action that saves the current state of our transaction ledger in memory before making a call to our payments database to post a new customer charge. This happens when the `charge` endpoint of our payments service is hit:
+
+{{< img src="tracing/guide/custom_span/custom_span_2.png" alt="Analytics View" responsive="true" style="width:90%;">}}
+
+Note that the `http.request POST /charge/` span is taking a lot of time without having any direct child spans. This is a clue that this request requires further instrumentation to gain better insights into its behavior. Depending on the programming language you are using, you need to decorate your functions differently:
```

Comment:
@omris1 as a side note, we may want to use another part of the application.

Reply:
How about instead, we change the legend in in photoshop?

Reply:
Let's use our real `shop.ist` app. In case we'll change our code representation there.